Source/SwiftLintFramework/Rules/JoinedDefaultRule.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  JoinedDefaultRule.swift
+//  SwiftLint
+//
+//  Created by Ornithologist Coder on 8/3/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct JoinedDefaultParameterRule: ASTRule, ConfigurationProviderRule, OptInRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "joined_default_parameter",
+        name: "Joined Default Parameter",
+        description: "Discouraged explicit usage of the default separator.",
+        kind: .idiomatic,
+        nonTriggeringExamples: [
+            "let foo = bar.joined()",
+            "let foo = bar.joined(separator: \",\")",
+            "let foo = bar.joined(separator: toto)"
+        ],
+        triggeringExamples: [
+            "let foo = bar.joined(↓separator: \"\")",
+            "let foo = bar.filter(toto)\n" +
+            "             .joined(↓separator: \"\")"
+        ]
+    )
+
+    public func validate(file: File,
+                         kind: SwiftExpressionKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        guard
+            kind == .call,
+            dictionary.name?.hasSuffix(".joined") == true,
+            let defaultSeparatorOffset = defaultSeparatorOffset(dictionary: dictionary, file: file)
+            else {
+                return []
+        }
+
+        return [StyleViolation(ruleDescription: type(of: self).description,
+                               severity: configuration.severity,
+                               location: Location(file: file, byteOffset: defaultSeparatorOffset))]
+    }
+
+    private func defaultSeparatorOffset(dictionary: [String: SourceKitRepresentable], file: File) -> Int? {
+        guard
+            let bodyOffset = dictionary.bodyOffset,
+            let bodyLength = dictionary.bodyLength else { return nil }
+
+        let body = file.contents.bridge().substringWithByteRange(start: bodyOffset, length: bodyLength)
```

Comment:
what do you think about using the substructure instead?

```json
 {
 	"key.namelength": 10,
 	"key.nameoffset": 1,
 	"key.length": 25,
 	"key.substructure": [{
 		"key.namelength": 9,
 		"key.nameoffset": 12,
 		"key.length": 13,
 		"key.name": "separator",
 		"key.bodyoffset": 23,
 		"key.bodylength": 2,
 		"key.kind": "source.lang.swift.expr.argument",
 		"key.offset": 12
 	}],
 	"key.name": "bar.joined",
 	"key.bodyoffset": 12,
 	"key.bodylength": 13,
 	"key.kind": "source.lang.swift.expr.call",
 	"key.offset": 1
 }
```

It should be more reliable than the current implementation.